### PR TITLE
 frequence countung strategy change

### PR DESCRIPTION
**Pull request by LuxoftAKutsan**

### CODE REVIEW

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/interior_data_manager_impl.h [hunk 1]
```diff
@@ -95,6 +102,7 @@ class InteriorDataManagerImpl : public InteriorDataManager {
    */
   void UnsubscribeFromInteriorVehicleData(const std::string& module_type);
 
+  void ClearOldSubscriptionsHostory();
```
Comment (LitvinenkoIra) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/interior_data_manager_impl.h]: @LuxoftAKutsan typo, should be _ClearOldSubscriptionsHistory_

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/interior_data_manager.h [hunk 2]
```diff
@@ -57,7 +58,26 @@ class InteriorDataManager {
...
 
+
+    /**
+     * @brief OnDisablingRC process disable RC event. UNsubscribe from all modules
+     * and clear cache
+     */
   virtual void OnDisablingRC() = 0;
+
+  /**
+   * @brief StoreInteriorDataSubscriptionTime save information and time stamp of current interior data subscriptions
```
Comment (LitvinenkoIra) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/interior_data_manager.h]: @LuxoftAKutsan too long string

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/interior_data_manager.h [hunk 3]
```diff
@@ -57,7 +58,26 @@ class InteriorDataManager {
       plugins::ApplicationEvent event,
       app_mngr::ApplicationSharedPtr application) = 0;
 
+
+    /**
+     * @brief OnDisablingRC process disable RC event. UNsubscribe from all modules
```
Comment (LitvinenkoIra) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/interior_data_manager.h]: @LuxoftAKutsan  _UNsubscribe -> Unsubscribe_

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/interior_data_manager_impl.h (hunk 1 above)
Comment (LuxoftAKutsan) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/interior_data_manager_impl.h]: 06485260a2

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/interior_data_manager.h (hunk 2 above)
Comment (LuxoftAKutsan) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/interior_data_manager.h]: 06485260a2

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/interior_data_manager.h (hunk 3 above)
Comment (LuxoftAKutsan) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/interior_data_manager.h]: @LitvinenkoIra fixed

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/interior_data_manager_impl.h [hunk 4]
```diff
@@ -52,25 +54,18 @@ class InteriorDataManagerImpl : public InteriorDataManager {
...
   void OnApplicationEvent(plugins::ApplicationEvent event,
                           app_mngr::ApplicationSharedPtr application) OVERRIDE;
 
-  /**
-   * @brief OnDisablingRC process disable RC event. UNsubscribe from all modules
-   * and clear cache
-   */
   void OnDisablingRC() OVERRIDE;
 
+  void StoreInteriorDataSubscriptionTime(
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/interior_data_manager_impl.h]: @alexkutsan rename to StoreRequestToHMITime

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/interior_data_manager_impl.h [hunk 5]
```diff
@@ -52,25 +54,18 @@ class InteriorDataManagerImpl : public InteriorDataManager {
...
+      const std::string& module_type) OVERRIDE;
+
+  bool CheckSubscriptionsFrequency(const std::string& module_type) OVERRIDE;
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/interior_data_manager_impl.h]: @alexkutsan rename to CheckRequestsToHMIFrequency

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/interior_data_cache_impl.cc [hunk 6]
```diff
@@ -42,20 +42,9 @@ namespace rc_rpc_plugin {
...
 
 InteriorDataCacheImpl::InteriorDataCacheImpl(
-    const uint32_t time_frame_of_allowed_requests)
-    : reset_request_count_timer_(
-          "InteriorDataCacheImplResetRequests",
-          new timer::TimerTaskImpl<InteriorDataCacheImpl>(
-              this, &InteriorDataCacheImpl::ResetRequestCountOnTimer)) {
-  StartRequestResetTimer(time_frame_of_allowed_requests);
-}
+    const uint32_t time_frame_of_allowed_requests) {}
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/interior_data_cache_impl.cc]: @alexkutsan  remove time_frame_of_allowed_requests parameter

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/interior_data_manager_impl.cc [hunk 7]
```diff
@@ -48,6 +48,26 @@ void InteriorDataManagerImpl::OnDisablingRC() {
...
+
+bool InteriorDataManagerImpl::CheckSubscriptionsFrequency(
+    const std::string& module_type) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  sync_primitives::AutoLock autolock(subscriptions_history_lock_);
+  ClearOldSubscriptionsHistory();
+  const auto& history = subscriptions_history_[module_type];
+  const auto limit =
+      app_mngr_.get_settings().get_interior_vehicle_data_frequency().first;
+  LOG4CXX_DEBUG(logger_, "History size " << history.size());
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/interior_data_manager_impl.cc]: @alexkutsan  remove this debug

**Pull request merged**